expediagroup-sdk-rest/src/main/kotlin/com/expediagroup/sdk/rest/util/UrlQueryParamUtils.kt

```diff
@@ -32,53 +32,134 @@ fun interface StringifyQueryParam : (UrlQueryParam) -> String {
+        query: String
+    ): UrlQueryParam
+}
 
-/**
- * Converts a UrlQueryParam to an exploded form-encoded String.
- * Example: key=value1&key=value2
- */
-val stringifyExplode =
-    StringifyQueryParam { param ->
-        StringBuilder().apply {
-            append("${param.key}=")
-            append(param.value.joinToString("&${param.key}="))
-        }.toString()
-    }
+class UrlQueryParamStringifiers {
+    companion object {
+        /**
+         * Converts a UrlQueryParam to a form-encoded String.
+         * Example: key=value1,value2
+         */
+        val stringifyForm =
+            StringifyQueryParam { param ->
+                StringBuilder().apply {
+                    append("${param.key}=")
+                    append(param.value.joinToString(URLEncoder.encode(",", "UTF-8")))
+                }.toString()
+            }
 
-/**
- * Converts a UrlQueryParam to a space-delimited String.
- * Example: key=value1%20value2
- */
-val stringifySpaceDelimited =
-    StringifyQueryParam { param ->
-        StringBuilder().apply {
-            append("${param.key}=")
-            append(param.value.joinToString(URLEncoder.encode(" ", "UTF-8")))
-        }.toString()
+        /**
+         * Converts a UrlQueryParam to an exploded form-encoded String.
+         * Example: key=value1&key=value2
+         */
+        val stringifyExplode =
+            StringifyQueryParam { param ->
+                StringBuilder().apply {
+                    append("${param.key}=")
+                    append(param.value.joinToString("&${param.key}="))
+                }.toString()
+            }
+
+        /**
+         * Converts a UrlQueryParam to a space-delimited String.
+         * Example: key=value1%20value2
+         */
+        val stringifySpaceDelimited =
+            StringifyQueryParam { param ->
+                StringBuilder().apply {
+                    append("${param.key}=")
+                    append(param.value.joinToString(URLEncoder.encode(" ", "UTF-8")))
+                }.toString()
+            }
+
+        /**
+         * Converts a UrlQueryParam to a pipe-delimited String.
+         * Example: key=value1|value2
+         */
+        val stringifyPipeDelimited =
+            StringifyQueryParam { param ->
+                StringBuilder().apply {
+                    append("${param.key}=")
+                    append(param.value.joinToString(URLEncoder.encode("|", "UTF-8")))
+                }.toString()
+            }
     }
+}
 
-/**
- * Converts a UrlQueryParam to a pipe-delimited String.
- * Example: key=value1|value2
- */
-val stringifyPipeDelimited =
-    StringifyQueryParam { param ->
-        StringBuilder().apply {
-            append("${param.key}=")
-            append(param.value.joinToString(URLEncoder.encode("|", "UTF-8")))
-        }.toString()
+class UrlQueryParamDestringifiers {
+    companion object {
+        /**
+         * Destringifies a form-encoded query string into a list of values.
```

Comment:
```suggestion
         * Parses a single - form-encoded - query param into its respective data type.
```
Can we also modify the rest of the functions below?

Reply:
Done.